tools/build_variables.bzl

```
@@ -314,7 +314,26 @@ core_sources_full = core_sources_full_mobile + [
 
 libtorch_core_sources = sorted(core_sources_common + core_sources_full + core_trainer_sources)
 
-libtorch_distributed_sources = [
+libtorch_distributed_windows_sources = [
+    "torch/lib/c10d/comm.cpp",
+    "torch/lib/c10d/default_comm_hooks.cpp",
+    "torch/lib/c10d/FileStore.cpp",
+    "torch/lib/c10d/GlooDeviceFactory.cpp",
+    "torch/lib/c10d/logger.cpp",
+    "torch/lib/c10d/ParamCommsUtils.cpp",
+    "torch/lib/c10d/PrefixStore.cpp",
+    "torch/lib/c10d/ProcessGroup.cpp",
+    "torch/lib/c10d/ProcessGroupGloo.cpp",
+    "torch/lib/c10d/ProcessGroupMPI.cpp",
+    "torch/lib/c10d/ProcessGroupWrapper.cpp",
+    "torch/lib/c10d/reducer.cpp",
+    "torch/lib/c10d/sequence_num.cpp",
+    "torch/lib/c10d/Store.cpp",
+    "torch/lib/c10d/TCPStore.cpp",
+    "torch/lib/c10d/Utils.cpp",
+]
+
+libtorch_distributed_sources = libtorch_distributed_windows_sources + [
```

If filelists are split into base and c10d source if we replace `if();add();else();add();endif()` into `add(base);if();add(extra);endif();`
```suggestion
libtorch_distributed_c10d_sources =  [
```

---

I don't want to use `c10d` in the name because after this change we'll be able to also move the RPC layer from `libtorch_python` to `libtorch` and its files would be added to this list as well. But I'm fine with calling it `extra` as suggested.

---

How about `libtorch_distributed_extra_sources`

---

Yes that's what I went with. I also left `libtorch_distributed_sources` defined as the union of base+extra, since it's handy internally. Hope this is fine?